### PR TITLE
Implement a sync version of the schedule and sensor grpc call

### PR DESCRIPTION
**Pull request by gibsondan**
## Summary & Motivation
We've never needed this to be an async call and we seem to be hitting upstream grpc issues with streaming API calls where they sporadically return early without raising an error. Make a sync version of the call and use that. For back-compat, fall back to the async version if an UnimplementedError is returned. 

This adds a marginal bit of latency in the case where you update your webserver/daemon/dagster cloud agent before you upgrade your code, but shouldn't increase the throughput on the boxes since the first call returns more or less immediately with an UnimplementedError without doing any actual work. If that's a concern we could try to add some caching of the fact that a particular server doesn't have the method implemented, but my sense is that that is more trouble than it is worth.

## How I Tested These Changes
BK, test back-compat case manually by running a sensor locally on a new version of dagster-webserver but an old version of `dagster api grpc'

### CODE REVIEW

**Comment by prha** [comment]
Would this be able to handle the case where a tick requests a very large number of runs?  I have a vague memory of that being an issue, and I can't remember if the streaming API handled extremely large responses in a more stable way.

File: python_modules/dagster/dagster/_grpc/client.py [hunk 1]
```diff
@@ -367,6 +367,12 @@ def streaming_external_repository(
                 "serialized_external_repository_chunk": res.serialized_external_repository_chunk,
             }
 
+    def _is_unimplemented_error(self, e: Exception):
+        return (
+            isinstance(e.__cause__, grpc.RpcError)
+            and cast(grpc.RpcError, e.__cause__).code() == grpc.StatusCode.UNIMPLEMENTED
+        )
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_grpc/client.py:374]: how confident are we that this is a stable detection scheme? anything in there docs confirm?

File: python_modules/dagster/dagster/_grpc/client.py [hunk 2]
```diff
@@ -376,18 +382,31 @@ def external_schedule_execution(
...
+                    self._streaming_query(
+                        "ExternalScheduleExecution",
+                        api_pb2.ExternalScheduleExecutionRequest,
+                        serialized_external_schedule_execution_args=serialize_value(
+                            external_schedule_execution_args
+                        ),
+                        timeout=timeout,
+                    )
+                )
+                return "".join([chunk.serialized_chunk for chunk in chunks])
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_grpc/client.py:407]: nit: unclear how much this is worth worrying about, but if we eat an exception while in this `except` block we'll get some additional exception noise that maybe advocates for trying to do this work outside of the `except` block? 

Eh I dunno, its probably fine.

**Review by alangenfeld — COMMENTED** [review]
Do we need any special considerations for proxy servers? Are they guaranteed to be in version sync with their underlying code server?

**Comment by gibsondan** [comment]
proxy servers open grpc servers as subprocesses so no versioning issues to worry about there

File: python_modules/dagster/dagster/_grpc/client.py (hunk 1 above)
Comment (gibsondan) [reply python_modules/dagster/dagster/_grpc/client.py:374]: we use it elsewhere - these are all part of the public / top-level grpc API and I think the StatusCodes have stayed stable as long as we've been using grpc

If they moved or renamed them i think a test failure would pop up quickly, i think moving UNIMPLEMENTED to some other verb is unlikely

**Comment by gibsondan** [comment]
@prha i think now that we are compressing and have a clearer understanding of what knobs you can pull to e.g. increase max response size that it is unlikely to be a problem in practice, RunRequests are typically not very large and we allow 50MB responses

**Review by alangenfeld — APPROVED** [review]

**Pull request merged**